File: android/src/main/kotlin/com/example/pauza_screen_time/core/method_channel/CoreMethodHandler.kt
package com.example.pauza_screen_time.core.method_channel

import com.example.pauza_screen_time.core.MethodNames
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class CoreMethodHandler : MethodCallHandler {
    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            MethodNames.GET_PLATFORM_VERSION -> {
                result.success("Android ${android.os.Build.VERSION.RELEASE}")
            }
            else -> result.notImplemented()
        }
    }
}

File: ios/Classes/Core/MethodChannel/CoreMethodHandler.swift
import Flutter
import UIKit

final class CoreMethodHandler {
    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case MethodNames.getPlatformVersion:
            result("iOS " + UIDevice.current.systemVersion)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
